# Data analysis. 
In this notebook I will collect information about the seasons 14-15 to 19-20 combined.  
I will answer those questions : 
* Which 5 teams won the most games overall ? 
* Which 5 teams lost the most games overall ? 
* Which 5 teams won the most games away 
* which 5 teams won the most games at home ? 
* Which 5 teams lost the most games away ? 
* Which 5 teams lost the most games at home ?
* Overall percentage of home wins ? 
* Overall percentage of away wins ? 
* Overall percentage of home losses ? 
* Overall percentage of away losses ? 
* Overall percentage of wins when trailing at half-time ? 
* Overall percetage of wins when leading at half-time ? 
* Overall percentage of losses when trailing at half-time ? 
* Overall percentage of losses when leading at half-time ? 
* Correlation between playing at home and winning the game ? 
* Correlation between playing away and winning the game ? 
* Correlation between the overall odds and the ftr ? 


## Import datasets and libraries

In [13]:
import pandas as pd
import numpy as np
import seaborn as sns

In [14]:
df = pd.read_csv("../assets/data/clean_data.csv")
df.drop("Unnamed: 0", inplace=True, axis=1)

## Which 5 teams won or lost the most games ? Away ? Home ? 

#### Feature engineering
Creation of new columns to easily count the losses and wins of each teams.

##### Creation of a full time winner column

In [44]:
def get_winner(ftr):
    if ftr.item() == "H":
        return df["HomeTeam"].iloc[ftr.name]
    elif ftr.item() == "A":
        return df["AwayTeam"].iloc[ftr.name]
    else :
        return "Draw"
df["ftr_winner"] = df[["FTR"]].apply(get_winner, axis=1)

##### Creation of a full time looser column

In [48]:
def get_looser(ftr):
    if ftr.item() == "H":
        return df["AwayTeam"].iloc[ftr.name]
    elif ftr.item() == "A":
        return df["HomeTeam"].iloc[ftr.name]
    else :
        return "Draw"
df["ftr_looser"] = df[["FTR"]].apply(get_looser, axis=1)

#### Value count 

##### Most wins overall

In [56]:
most_wins = df["ftr_winner"].value_counts(sort=True)
most_wins.head(6)

Draw         547
Man City     156
Liverpool    139
Chelsea      130
Tottenham    126
Arsenal      119
Name: ftr_winner, dtype: int64

##### Most losses overall 

In [58]:
most_losses = df["ftr_looser"].value_counts(sort=True)
most_losses.head(6)

Draw              547
Crystal Palace    105
Watford            92
West Ham           91
Bournemouth        91
Southampton        90
Name: ftr_looser, dtype: int64

##### Most home wins 

In [62]:
home_wins = df.loc[df["FTR"]== "H"]
most_home_wins = home_wins["ftr_winner"].value_counts(sort=True)
most_home_wins.head(5)

Man City     86
Arsenal      77
Liverpool    77
Tottenham    74
Chelsea      71
Name: ftr_winner, dtype: int64

##### Most away wins

In [70]:
away_wins = df.loc[df["FTR"]== "A"]
most_away_wins = away_wins["AwayTeam"].value_counts(sort=True)
most_away_wins.head(5)

Man City      70
Liverpool     62
Chelsea       59
Tottenham     52
Man United    50
Name: AwayTeam, dtype: int64

##### Most home losses

In [73]:
home_losses = df.loc[df["FTR"] == "A"]
most_home_losses = home_losses["HomeTeam"].value_counts(sort=True)
most_home_losses.head(5)

Crystal Palace    55
Southampton       40
West Ham          38
Burnley           38
Bournemouth       36
Name: HomeTeam, dtype: int64

##### Most away losses 

In [75]:
away_losses = df.loc[df["FTR"] == "H"]
most_away_losses = away_losses["AwayTeam"].value_counts(sort=True)
most_away_losses.head(5)

Watford        57
Newcastle      55
Bournemouth    55
Everton        54
West Ham       53
Name: AwayTeam, dtype: int64

## Overall percentage home wins, away wins, home losses, away losses

##### Percentage of home wins and away wins

In [83]:
percentages = df["FTR"].value_counts(normalize=True) *100
percentages = percentages.to_frame()
percentages

FTR
H  45.701754
A  30.350877
D  23.947368

Home wins percentage  : 45.70 %.   

Away win percentage : 30.35 %.   

The percentages for the home losses and away losses are the same but inverted obviously.  

Away losses percentage : 45.70%.  

Home losses  percentage : 30.35%

## Wins or losses when trailing or leading at half time